Sources/SwiftBSON/Array+BSONValue.swift

```diff
@@ -46,7 +46,12 @@ extension Array: BSONValue where Element == BSON {
         guard let doc = try BSONDocument.read(from: &buffer).documentValue else {
             throw BSONError.InternalError(message: "BSON Array cannot be read, failed to get documentValue")
         }
-        return .array(doc.values)
+        var values: [BSON] = []
+        let it = doc.makeIterator()
+        while let (_, val) = try it.nextThrowing() {
```

Comment:
since `BSONDocument.read` doesn't validate anymore, we need to use `nextThrowing` to ensure we don't `fatalError`. Full validation will be performed in `Array.validate` if necessary.